packages/core/src/shared/telemetry/telemetryLogger.ts

```diff
@@ -118,4 +118,13 @@ export class TelemetryLogger {
     public queryFull(query: MetricQuery): MetricDatum[] {
         return this._metrics.filter((m) => m.MetricName === query.metricName)
     }
+
+    /**
+     * Queries telemetry for metrics with metadata containing a keyword in the key or value
+     */
+    public queryKeyword(keyword: string): MetricDatum[] {
```

Comment:
might as well support regex?
```suggestion
    public queryRegex(re: RegExp): MetricDatum[] {
```